packages/core/src/file-upload/file-upload-hidden-input.tsx

```diff
@@ -0,0 +1,51 @@
+import type { JSX, ValidComponent } from "solid-js";
+import {
+	type ElementOf,
+	Polymorphic,
+	type PolymorphicProps,
+} from "../polymorphic";
+import { useFileUploadContext } from "./file-upload-root-provider";
+
+export type FileUploadHiddenInputCommonProps<
+	T extends HTMLElement = HTMLElement,
+> = {
+	id?: string;
+	style?: JSX.CSSProperties | string;
+};
+
+export type FileUploadHiddenInputRootProps<
+	T extends ValidComponent | HTMLElement = HTMLElement,
+> = Partial<FileUploadHiddenInputCommonProps<ElementOf<T>>>;
+
+export function FileUploadHiddenInput<T extends ValidComponent = "input">(
+	props: PolymorphicProps<T, FileUploadHiddenInputRootProps<T>>,
+) {
+	const context = useFileUploadContext();
+
+	const onInput: JSX.EventHandler<HTMLInputElement, InputEvent> = (event) => {
+		if (context.disabled) {
+			return;
+		}
+
+		const { files } = event.currentTarget;
+		console.log("files", files);
+		const t = Array.from(files ?? []);
+		context.processFiles(Array.from(files ?? []));
+	};
+
+	return (
+		<Polymorphic
+			as="input"
+			type="file"
+			id={context.inputId}
+			accept={context.accept}
+			multiple={context.multiple}
+			ref={(el: HTMLInputElement) => (context.fileInputRef = el)}
+			style={{ display: "none" }}
```

Comment:
For hiding an input please use `visuallyHiddenStyles` from `@kobalte/utils` ([example here](https://github.com/kobaltedev/kobalte/blob/76319603b4b5242c2533f1cc5e910a3d9cd51d55/packages/core/src/switch/switch-input.tsx#L140)).

Reply:
Done